CONTRIBUTING.md

```diff
@@ -0,0 +1,183 @@
+maybe your question was already answered.
+
+If your issue appears to be a bug, and hasn't been reported, open a new issue.
+Help us to maximize the effort we can spend fixing issues and adding new
+features, by not reporting duplicate issues.  Providing the following information will increase the
+chances of your issue being dealt with quickly:
+
+* **Overview of the Issue** - what type of issue is it, and why is it an issue for you?
+* **Callstack** - if it's a crash or other runtime error, a callstack will help diagnosis
+* **Logs** - stdout and stderr from the console, or log files if there are any
+* **Reproduction steps** - a minimal, unambigious set of steps including input, that causes the error for you.
+    e.g. input geometry and settings you can use to input into RecastDemo to get it to fail.
+* **Recast version(s) and/or git commit hash** - particularly if you can find the point at which the error first started happening
+* **Environment** - operating system, compiler etc.
+* **Related issues** - has a similar issue been reported before?
+* **Suggest a Fix** - if you can't fix the bug yourself, perhaps you can point to what might be
+  causing the problem (line of code or commit)
+
+Here is a great example of a well defined issue: https://github.com/recastnavigation/recastnavigation/issues/12
+
+**If you get help, help others. Good karma rulez!**
+
+### Submitting a Pull Request
+Before you submit your pull request consider the following guidelines:
+
+* Search [GitHub Pull Requests][github-pulls] for an open or closed Pull Request
+  that relates to your submission. You don't want to duplicate effort.
+* Make your changes in a new git branch:
+
+     ```shell
+     git checkout -b my-fix-branch master
+     ```
+
+* Implement your changes, **including appropriate tests if appropriate/possible**.
+* Commit your changes using a descriptive commit message that follows our
+  [commit message conventions](#commit-message-format).
+
+     ```shell
+     git commit -a
+     ```
+  Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
+
+* Squash any work-in-progress commits (by rebasing) to form a series of commits that make sense individually.
+  Ideally the pull request will be small and focused enough that it fits sensibly in one commit.
+
+     ```shell
+     git rebase -i origin/master
+     ```
+
+* Push your branch to GitHub:
+
+    ```shell
+    git push origin my-fix-branch
+    ```
+
+* In GitHub, send a pull request to `recastnavigation:master`.
+* If we suggest changes then:
+  * Make the required updates.
+  * Commit your changes to your branch (e.g. `my-fix-branch`).
+  * Squash the changes, overwriting history in your fix branch - we don't want history to include incomplete work.
+  * Push the changes to your GitHub repository (this will update your Pull Request).
+
+You will need to force the push if you rebased.

```

Comment:
This looks like a duplicate line? The line below says the same thing with more detail.